### PR TITLE
Memory leak in custom batch-norm

### PR DESCRIPTION
**Pull request by holynski**
Using the `ResNetEncoder` and `ResNetDecoder` models results in a pretty hefty (~7MB/s) memory leak on CPU memory, which ultimately results in training crashing after a couple days.

I traced it down to a few lines in the custom batch-norm. The variables `self.stored_mean`, `self.stored_var` seem to only be used during inference, but are not protected from computing gradients, and thus are extending the computation graph on each call to `bn.forward()`.

An old issue on the main PyTorch repo seems to validate these findings:
https://github.com/pytorch/pytorch/issues/20275

I've tested with this change, and the memory usage remains constant.

### CODE REVIEW

**Comment by jcjohnson** [comment]
This indeed looks like a bug on our part -- thanks for finding this and providing a fix!

@oawiles do you mind taking a quick look before I merge this?

**Comment by oawiles** [comment]
Yes I will look this evening!

**Comment by oawiles** [comment]
Yes, this seems good! And a very simple fix thanks @holynski

**Comment by holynski** [comment]
Sure, happy to help!
